Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -0,0 +1,598 @@
+      }
+
+      public ConcreteSyntaxTree CreateGetter(string name, TopLevelDecl enclosingDecl, Type resultType, IToken tok,
+        bool isStatic,
+        bool isConst, bool createBody, MemberDecl member, bool forBodyInheritance) {
+        return Compiler.CreateGetter(name, resultType, tok, isStatic, createBody, MethodWriter);
+      }
+
+      public ConcreteSyntaxTree CreateGetterSetter(string name, Type resultType, IToken tok, bool isStatic,
+        bool createBody,
+        MemberDecl member, out ConcreteSyntaxTree setterWriter, bool forBodyInheritance) {
+        return Compiler.CreateGetterSetter(name, resultType, tok, isStatic, createBody, out setterWriter, MethodWriter);
+      }
+
+      public void DeclareField(string name, TopLevelDecl enclosingDecl, bool isStatic, bool isConst, Type type,
+        IToken tok,
+        string rhs, Field field) {
+        Compiler.DeclareField(name, isStatic, isConst, type, tok, rhs, FieldWriter);
+      }
+
+      public void InitializeField(Field field, Type instantiatedFieldType, TopLevelDeclWithMembers enclosingClass) {
+        throw new NotSupportedException();
+      }
+
+      public ConcreteSyntaxTree ErrorWriter() => MethodWriter;
+
+      public void Finish() {
+
+      }
+    }
+
+    private void DeclareField(string name, bool isStatic, bool isConst, Type type, IToken tok, string rhs,
+      ConcreteSyntaxTree fieldWriter) {
+      throw new NotImplementedException();
+    }
+
+    private ConcreteSyntaxTree CreateGetterSetter(string name, Type resultType, IToken tok, bool isStatic,
+      bool createBody, out ConcreteSyntaxTree setterWriter, ConcreteSyntaxTree methodWriter) {
+      throw new NotImplementedException();
+    }
+
+    private ConcreteSyntaxTree CreateGetter(string name, Type resultType, IToken tok, bool isStatic, bool createBody,
+      ConcreteSyntaxTree methodWriter) {
+      throw new NotImplementedException();
+    }
+
+    private ConcreteSyntaxTree CreateMethod(Method m, List<TypeArgumentInstantiation> typeArgs, bool createBody,
+      ConcreteSyntaxTree wr, bool forBodyInheritance, bool lookasideBody) {
+      if (!createBody) {
+        return null;
+      }
+
+      var customReceiver = !forBodyInheritance && NeedsCustomReceiver(m);
+      wr.Write("{0}{1}(", m.IsStatic || customReceiver ? "def " : "", IdName(m));
+      var sep = "";
+      WriteRuntimeTypeDescriptorsFormals(m, ForTypeDescriptors(typeArgs, m, lookasideBody), wr, ref sep,
+        tp => $"rtd$_{tp.CompileName}");
+      if (customReceiver) {
+        var nt = m.EnclosingClass;
+        var receiverType = UserDefinedType.FromTopLevelDecl(m.tok, nt);
+        DeclareFormal(sep, "_this", receiverType, m.tok, true, wr);
+        sep = ", ";
+      }
+
+      WriteFormals(sep, m.Ins, wr);
+      var w = wr.NewBlock("):", open: BraceStyle.Pindent, close: BraceStyle.Pindent);
+
+      if (!m.IsStatic && !customReceiver) {
+        w.WriteLine("let _this = this;");
```

Comment:
This isn't valid Python - I assume it was just copied over from the Javascript version. You should be able to just omit this if entirely for now AFAICT.